cardano-testnet/src/Testnet/Start/Cardano.hs

```diff
@@ -55,41 +59,51 @@ import qualified Data.Time.Clock as DTC
 import           GHC.Stack
 import qualified System.Directory as IO
 import           System.FilePath ((</>))
-import qualified System.Info as OS
 
 import           Testnet.Components.Configuration
 import qualified Testnet.Defaults as Defaults
 import           Testnet.Filepath
 import           Testnet.Handlers (interruptNodesOnSigINT)
-import           Testnet.Process.Run (execCli', execCli_, mkExecConfig)
+import           Testnet.Process.RunIO (execCli', execCli_, liftIOAnnotated, mkExecConfig)
 import           Testnet.Property.Assert (assertChainExtended, assertExpectedSposInLedgerState)
 import           Testnet.Runtime as TR
 import           Testnet.Start.Types
 import           Testnet.Types as TR hiding (shelleyGenesis)
 
-import           Hedgehog (MonadTest)
-import qualified Hedgehog as H
 import qualified Hedgehog.Extras as H
 import qualified Hedgehog.Extras.Stock.IO.Network.Port as H
 
+
+import RIO (RIO(..),runRIO, throwString, MonadUnliftIO)
+import Control.Monad.Trans.Resource (getInternalState, MonadResource)
+import Testnet.Orphans ()
+import RIO.Orphans (ResourceMap)
+import UnliftIO.Async
+
 -- | There are certain conditions that need to be met in order to run
 -- a valid node cluster.
 testMinimumConfigurationRequirements :: ()
   => HasCallStack
-  => MonadTest m
+  => MonadIO m
   => CardanoTestnetOptions -> m ()
 testMinimumConfigurationRequirements options = withFrozenCallStack $ do
   when (cardanoNumPools options < 1) $ do
-    H.note_ "Need at least one SPO node to produce blocks, but got none."
-    H.failure
+    throwString "Need at least one SPO node to produce blocks, but got none."
+
+liftToIntegration :: RIO ResourceMap a -> H.Integration a 
+liftToIntegration  r = do 
+   rMap <- lift $ lift getInternalState 
+   liftIOAnnotated $ runRIO rMap r
```

Comment:
Shouldn't that be `evalIO` instead? Or at least `evalIO . liftIOAnnotated`?
```suggestion
   H.evalIO $ runRIO rMap r
```
I think regular `liftIO` (which inside `liftIOAnnotated`) was losing some location information for me when reporting in the hedgehog. I don't know if that's still the case here.